generate_docs.sh

```diff
@@ -13,22 +13,37 @@ set -e
 # Get current directory of script.
 source_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-skip_checks=false
-if [ "$#" -eq 1 -a "$1" == "--skip-checks" ]; then
-    skip_checks=true
+help_msg="
+Usage: ./generate_docs.sh [OPTIONS]
+1. Builds and installs the source
+2. Runs doxygen and create HTML and XML docs
+3. Runs script to generate markdown from XML
+
+  --clean    for clean build and removing install dir.
```

Comment:
This basically changes the default to a non-clean build which means this script will fail for anyone that has not done a clean build beforehand. The reason for this is that it needs a clean build in order for the install directory to change to a local one as required by the docs script.

Therefore, a user that does not know this will run the script without `--clean` and it will fail and they will be unhappy. My vote is to error on the annoying side and have the option `--skip-checks` or `--skip-clean` for anyone that knows what they are doing.

Reply:
Yes.. right!
We can do like this: 
Prompt user once to cleanup build & install directory (*with commands*) when ran without `--skip-checks`. Updated this in the latest commit.